experimental/python/wheel.bzl

```diff
@@ -0,0 +1,159 @@
+        extra_headers.append("Home-page: %s" % ctx.attr.homepage)
+    if ctx.attr.license:
+        extra_headers.append("License: %s" % ctx.attr.license)
+
+    for h in extra_headers:
+        arguments.append("--header")
+        arguments.append(h)
+
+    for c in ctx.attr.classifiers:
+        arguments.append("--classifier")
+        arguments.append(c)
+
+    for r in ctx.attr.requires:
+        arguments.append("--requires")
+        arguments.append(r)
+
+    for option, requirements in ctx.attr.extra_requires.items():
+        for r in requirements:
+            arguments.append("--extra_requires")
+            arguments.append(r + ";" + option)
+
+    for name, ref in ctx.attr.console_scripts.items():
+        arguments.append("--console_script")
+        arguments.append(name + " = " + ref)
+
+    if ctx.attr.description_file:
+        description_files = ctx.attr.description_file.files.to_list()
+        arguments.append("--description_file")
+        arguments.append(description_files[0].path)
+        inputs = inputs.union(ctx.attr.description_file.files)
+
+    ctx.actions.run(
+        inputs = inputs,
+        outputs = [outfile],
+        arguments = arguments,
+        executable = ctx.executable._wheelmaker,
+        progress_message = "Building wheel",
+    )
+    return [DefaultInfo(
+        files = depset([outfile]),
+        data_runfiles = ctx.runfiles(files = [outfile]),
+    )]
+
+py_wheel = rule(
+    implementation = _py_wheel_impl,
+    attrs = {
+        "deps": attr.label_list(
+            doc = "Dependencies of this package, e.g. py_library rules",
+        ),
+        "packages": attr.string_list(
+            mandatory = True,
+            allow_empty = False,
+            doc = """\
+List of Python packages to include in the distribution.
+Sub-packages are automatically included.
+""",
+        ),
+        # Attributes defining the distribution
+        "wheel_name": attr.string(
+            mandatory = True,
+            doc = "Name of the wheel",
+        ),
+        "version": attr.string(
+            mandatory = True,
+            doc = "Version number of the package",
+        ),
+        "python": attr.string(
+            default = "py3",
+            doc = "Supported Python major version. 'py2' or 'py3'",
+        ),
+        "abi": attr.string(
+            default = "none",
+            doc = "Python ABI tag. 'none' for pure-Python wheels.",
+        ),
+        # TODO(pstradomski): Support non-pure wheels
+        "platform": attr.string(
+            default = "any",
+            doc = "Supported platforms. 'any' for pure-Python wheel.",
+        ),
+        # Other attributes
+        "author": attr.string(default = ""),
+        "author_email": attr.string(default = ""),
+        "homepage": attr.string(default = ""),
+        "license": attr.string(default = ""),
+        "classifiers": attr.string_list(),
+        "description_file": attr.label(allow_files = True),
```

Comment:
If this logically can only ever be a single file, then use [`allow_single_file`](https://docs.bazel.build/versions/master/skylark/lib/attr.html#label.allow_single_file) instead of `allow_files`. That will enable you to use `ctx.file.description_file` in the rule implementation function.

Reply:
Done.